### PR TITLE
Improve meson information, structure and move fortunes

### LINKED ISSUE #210: Fortunes install directory
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| rizin -v full output, **not truncated** (mandatory)  | rizin 0.1.0-git 26141 @ linux-x86-64 git.4.4.0-1159-g3a94841da commit: 3a94841da363a635a1dd9fefc9409f060baec612 build: 2020-12-16__21:26:45

We provide `rizin_libdir`, `rizin_fortunes`, etc. options in meson, but they are considered only for Windows. Why? They can be useful for linux as well. For example, see https://github.com/rizinorg/rizin/issues/179#issuecomment-745476174 where gentoo requires a different doc path.

However, I question whether `/usr/share/doc/rizin/` is really the right place for fortunes, because they are not documentation files. Maybe they should be `/usr/share/rizin/fortunes/*`?

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/rizinbook) with the relevant information (if needed)

**Detailed description**

* Move fortunes from `/usr/share/doc/rizin` to `/usr/share/rizin/<version>/fortunes/`. I don't think fortunes are documentation files actually. If i look at my distro, I see there files like AUTHORS, Changelog, COPYING, etc. I think they better fit in `/usr/share/rizin/<version>` (maybe `/usr/share/rizin/fortunes` instead of `/usr/share/rizin/<version>/fortunes`?)
* Simplify a bit the handling of paths in windows path
* Allow to overwrite any directory for both windows and non-windows
* Use summary to display a better summary of what's going on in meson
* Increase minimum meson version to 0.55.0 (this is anyway required if we switch to meson subprojects)

**Test plan**

Nothing special, everything should still work.

**Closing issues**

Fix https://github.com/rizinorg/rizin/issues/210

**Example summary**

```
  Directories
                        prefix: /home/rschiron/.local
                        libdir: lib64
                    includedir: include/librz
                       datadir: share
                       wwwroot: share/rizin/0.1.0-git/www
                           sdb: share/rizin/0.1.0-git
                         zigns: share/rizin/0.1.0-git/zigns
                        themes: share/rizin/0.1.0-git/cons
                         flags: share/rizin/0.1.0-git/flag
                           hud: share/rizin/0.1.0-git/hud
                       plugins: lib64/rizin/0.1.0-git
                        extras: lib64/rizin-extras/0.1.0-git
                      bindings: lib64/rizin-bindings/0.1.0-git

  Configuration
              Debugger enabled: YES
          System magic library: YES
         System xxhash library: YES
        System openssl library: YES
          System libuv library: YES
       System capstone library: YES
    System tree-sitter library: NO
            System lz4 library: YES
           System zlib library: YES
            System zip library: YES
               Use ptrace-wrap: YES
                     Use RPATH: YES

  Plugins
                  Core Plugins: a2f, java
              Analysis Plugins: 6502, 8051, amd29k, arc, arm_cs, arm_gnu, avr, bf, chip8, cr16, cris, dalvik, ebc, gb, h8300, hexagon, i4004, i8080, java, m68k_cs, m680x_cs, malbolge, mcore, mips_cs, mips_gnu, msp430, nios2, null, or1k, pic, ppc_cs, ppc_gnu, propeller, riscv, rsp, sh, snes, sparc_cs, sparc_gnu, spc700, sysz, tms320, tricore, v810, v850, vax, wasm, ws, x86_cs, xap, xcore_cs, xtensa, z80
             Assembler Plugins: 6502, 8051, amd29k, arc, arm_as, arm_cs, arm_gnu, arm_winedbg, avr, bf, chip8, cr16, cris_gnu, dalvik, dcpu16, ebc, gb, h8300, hexagon, hppa_gnu, i4004, i8080, java, lanai_gnu, lh5801, lm32, m68k_cs, m680x_cs, malbolge, mcore, mcs96, mips_cs, mips_gnu, msp430, nios2, or1k, pic, ppc_as, ppc_cs, ppc_gnu, propeller, riscv, rsp, sh, snes, sparc_cs, sparc_gnu, spc700, sysz, tms320, tms320c64x, tricore, v810, v850, vax, wasm, ws, x86_as, x86_cs, x86_nasm, x86_nz, xap, xcore_cs, xtensa, z80
             Breapoint Plugins: arm, bf, mips, ppc, sh, x86
                Binary Plugins: any, art, avr, bf, bflt, bios, bootimg, cgc, coff, dex, dmp64, dol, dyldcache, elf, elf64, java, le, mach0, mach064, mbn, mdmp, menuet, mz, ne, nes, nin3ds, ninds, ningb, ningba, nro, nso, omf, qnx, p9, pe, pe64, pebble, prg, psxexe, sfc, smd, sms, spc700, symbols, te, vsf, wasm, xbe, xnu_kernelcache, z64, zimg
                BinLdr Plugins: ldr_linux
                BinXtr Plugins: xtr_dyldcache, xtr_fatmach0, xtr_sep64
                Crypto Plugins: aes, aes_cbc, base64, base91, blowfish, cps2, des, punycode, rc2, rc4, rc6, rol, ror, rot, serpent, xor
                    IO Plugins: ar, fd, bfdbg, bochs, debug, default, gdb, gprobe, gzip, http, ihex, mach, malloc, mmap, null, procpid, ptrace, qnx, rzk, rzpipe, rzweb, rap, rbuf, self, shm, sparse, tcp, winkd, winedbg, zip
                 Debug Plugins: bf, bochs, esil, gdb, io, native, null, qnx, rap, winkd
                   Egg Plugins: exec, xor
                  Lang Plugins: lib, pipe, rust, vala, c, cpipe
                 Parse Plugins: 6502_pseudo, arm_pseudo, att2intel, avr_pseudo, chip8_pseudo, tms320_pseudo, dalvik_pseudo, m68k_pseudo, mips_pseudo, ppc_pseudo, sh_pseudo, riscv_pseudo, v850_pseudo, wasm_pseudo, x86_pseudo, z80_pseudo
```

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -716,3 +667,51 @@ if cli_enabled
...
+  'System zip library': zip_dep.type_name() != 'internal',
+  'Use ptrace-wrap': use_ptrace_wrap,
+  'Use RPATH': use_rpath,
+}, section: 'Configuration', bool_yn: true)
+
+summary({
+  'Core Plugins': core_plugins,
+  'Analysis Plugins': analysis_plugins,
+  'Assembler Plugins': asm_plugins,
+  'Breapoint Plugins': bp_plugins,
```
Comment (Maijin) [inline meson.build]: Typo.
Also, any reason not to sort those summary alphabetically?

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
Looks cleaner now.

**Comment by XVilka** [comment]
For some reason "Test Windows installer built with clang_cl (pull_request)" job stuck at "Install rizin.exe" stage.

**Review by XVilka — APPROVED** [review]
Ready to merge now, imho.

**Pull request merged**